hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/sql/SqlEntityListener.java

```diff
@@ -317,13 +345,12 @@ public void onCustomFee(CustomFee customFee) throws ImporterException {
 
     @Override
     public void onEntity(Entity entity) throws ImporterException {
-        EntityId entityId = entity.toEntityId();
-        // handle empty id case
-        if (EntityId.isEmpty(entityId)) {
+        long id = entity.getId();
+        if (id == EntityId.EMPTY.getId() || contracts.containsKey(id)) {
```

Comment:
Do we reliably know the Entity type? If so another way we could do it is below instead
```suggestion
        if (id == EntityId.EMPTY.getId() || entity.getType() == EntityTypeEnum.CONTRACT.getId()) {
```

Reply:
We definitely don't know the type reliably. A crypto transfer for an contract could be sent to the `EntityListener` first and will show up as an account.